src/components/MasterItemsSelect.js

```diff
@@ -6,8 +6,10 @@ class MasterItemsSelect extends Component {
     super(props);
     this.handleChange = this.handleChange.bind(this);
     this.handleChangeCategory = this.handleChangeCategory.bind(this);
+    const filteredItems = this.filterItems(this.props.masterItems[0].category);
     this.state = {
-      filteredItems: this.filterItems(this.props.masterItems[0].category)
+      filteredItems,
+      selectedItemId: filteredItems[0].id
```

Comment:
The bug was there before...i never noticed it because we only had one item in the other category. The bug happens when there are multiple items in each category. If you select item2 in category1, and change to category2, the display value stays on item2's position even when on category2, but item1 category2 is sent to onSelectMasterItem (see line 34 - MasterItemSelect). i added the value attribute so the display value, and what's set in onSelectMasterItem is the same item. There has to be a better way to do this...